### PR TITLE
[NUI]  Public APIs Copy, Cut, Paste and SelectText

### PR DESCRIPTION
**Pull request by shrouqsabah**
### Change Description ###

Public APIs Copy, Cut, Paste and SelectedText in TextField or  TextEditor

According to ACR request.
We need to explain how to use public APIs Copy, Cut, Paste and SelectedText in TextField or TextEditor.

Added at the end of docs in "**docs/application/dotnet/guides/user-interface/nui/text.md**" a header.
The header is "**Use public APIs Copy, Cut, Paste and SelectedText in TextField or TextEditor**"
Explained APIs and added sequence of steps how to use them together.

For instance,

 - Issue #1735

### API Changes ###

For instance,

 - ACR: ACR-455

### CODE REVIEW

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (omar-srbd) [inline docs/application/dotnet/guides/user-interface/nui/text.md:759]:
```suggestion
### Use public APIs copy, cut, paste and select text in TextField or TextEditor
```

**Review by omar-srbd — COMMENTED** [review]
Hello @shrouqsabah, I hope you are doing well.
I have checked the parts from your commit and left some correction suggestions.
Please incorporate them at your earliest convenience.
Thank you.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (omar-srbd) [inline docs/application/dotnet/guides/user-interface/nui/text.md:793]: Please revert this change to `TextUtils.CopyToClipboard`, I had a misconception about the API.
```suggestion
`TextUtils.CopyToClipboard` is an API that takes one parameter, either TextField or TextEditor.
```

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (omar-srbd) [inline docs/application/dotnet/guides/user-interface/nui/text.md:807]: Please revert this change to `TextUtils.PasteTo`, I had a misconception about the API.
```suggestion
`TextUtils.PasteTo` is an API that takes one parameter, either TextField or TextEditor.
```

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (omar-srbd) [inline docs/application/dotnet/guides/user-interface/nui/text.md:826]: Please revert this change to `TextUtils.CutToClipboard`, I had a misconception about the API.
```suggestion
`TextUtils.CutToClipboard` is an API that takes one parameter, either TextField or TextEditor.
```

**Review by omar-srbd — COMMENTED** [review]
Hello @shrouqsabah, thanks for your most recent commit implementing the changes.
After another check of the file I have found a few more issues that need to be addressed, please incorporate the changes at your earliest convenience.
Thanks.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

**Review by omar-srbd — APPROVED** [review]
@shrouqsabah, thanks for incorporating all the changes as per the suggestion.

@asm-foysal @mijin85cho, I am approving this PR for merging.

**Comment by asm-foysal** [comment]
@omar-srbd is the heading count right? This file looks like it should be a second heading (at the same level as TextField and TextEditor) since it's reflecting data for both those sections... Also, the heading name seems too big, could we shorten it somehow?
@mijin85cho what do you think?

**Comment by mijin85cho** [comment]
> @omar-srbd is the heading count right? This file looks like it should be a second heading (at the same level as TextField and TextEditor) since it's reflecting data for both those sections... Also, the heading name seems too big, could we shorten it somehow? @mijin85cho what do you think?


I agree to your comments.

1. According to the context, this new paragraph would be much better to use 2nd heading.

2.The title would be better to shorten if possible. Concise title is recommended.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (omar-srbd) [inline docs/application/dotnet/guides/user-interface/nui/text.md:759]: Hello @shrouqsabah, I had a query regarding the heading.
The heading here seems to be a bit too long, we recommend using something concise for headings.

In this case, could you please clarify if the heading should emphasize the functionality of **Copy, cut, paste, and select text** or the usage of **public APIs**? Depending on your clarification I can provide an alternate for the heading or it would be better if you could change the heading to something concise.

Thank you.

Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md:759]: Hello @omar-srbd ,
We need to explain how to use APIs.
Here's the original request from ACR. 
**TCSACR-455** : ` Please add how to use new APIs. When you update the API guides, it will be good if you add how to use SelectedText API `.

Let's me know what you suggest :)
Thanks

Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md:759]: @omar-srbd  What do you think about this ( **How to use Clipboard APIs with Select Text** )?

Comment (omar-srbd) [inline docs/application/dotnet/guides/user-interface/nui/text.md:759]:
```suggestion
## How to use clipboard APIs with select text
```

**Review by omar-srbd — COMMENTED** [review]
Hello @shrouqsabah,

Please update the heading accordingly.

Thanks you.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (omar-srbd) [reply docs/application/dotnet/guides/user-interface/nui/text.md:759]: @shrouqsabah,

I have left another review incorporating your suggestion for the heading.
Please look into it and update accordingly.

Thanks.

Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (omar-srbd) [inline docs/application/dotnet/guides/user-interface/nui/text.md:783]:
```suggestion
// Step 01: Select text from source text field
```

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (omar-srbd) [inline docs/application/dotnet/guides/user-interface/nui/text.md:797]:
```suggestion
// Step 02: Copy text from source text field
```

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (omar-srbd) [inline docs/application/dotnet/guides/user-interface/nui/text.md:816]:
```suggestion
// Step 03: Paste text into destination text field
```

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (omar-srbd) [inline docs/application/dotnet/guides/user-interface/nui/text.md:830]:
```suggestion
// Step 04: Select text from source text field
```

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (omar-srbd) [inline docs/application/dotnet/guides/user-interface/nui/text.md:831]: I have a doubt regarding the selected text based on the declared indexes (0,8).
According to your declared indexes, the selected text should not include a **space** after the **exclamation mark (!)**.
Can you please check this?
```suggestion
textFieldSrc.SelectText (0,7); // "Welcome!" is the selected text
```

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (omar-srbd) [inline docs/application/dotnet/guides/user-interface/nui/text.md:840]:
```suggestion
// Step 05: Cut text from source text field
```

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (omar-srbd) [inline docs/application/dotnet/guides/user-interface/nui/text.md:849]:
```suggestion
// Step 06: Paste text into destination text field
```

**Review by omar-srbd — COMMENTED** [review]
@shrouqsabah, after another review I found a few more issues that need to be addressed and left correction suggestions for them.
Please incorporate them at your convenience.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md:835]: Checked.
I added the below note for description of SelectText. And fixed the first example:
`
   > [!NOTE]
   > The selected text includes the start index.
   > But, the selected text does not include the end index.

textFieldSrc.SelectText (9,16); //"This is" is the selected text
`

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: Done.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (omar-srbd) [inline docs/application/dotnet/guides/user-interface/nui/text.md:782]:
```suggestion
   > The selected text includes the start index, but it does not include the end index.
```

**Review by omar-srbd — COMMENTED** [review]
@shrouqsabah, after a review of your last commit, I found one issue that needs to be addressed.
Please check the suggestion and incorporate the change accordingly.
Thanks.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md]: @omar-srbd ,
Done. Thank you,

**Review by omar-srbd — APPROVED** [review]
@shrouqsabah, thanks for incorporating all the suggested changes.

@mijin85cho @asm-foysal I am approving this PR for merging.

File: docs/application/dotnet/guides/user-interface/nui/text.md
Comment (safir-srbd) [inline docs/application/dotnet/guides/user-interface/nui/text.md:759]: @shrouqsabah Hello. Hope you are well!

After checking before merging, I have a query regarding this PR.

In the heading, you have mentioned `SelectedText`, but in your changes here, you have written `SelectText` API. From what I have observed, `SelectText` is the appropriate name, as mentioned [here](https://docs.tizen.org/application/dotnet/api/TizenFX/latest/api/Tizen.NUI.BaseComponents.TextField.html#Tizen_NUI_BaseComponents_TextField_SelectText_System_Int32_System_Int32_). Can you please confirm if it should be `SelectedText` API or `SelectText` API?

Comment (shrouqsabah) [reply docs/application/dotnet/guides/user-interface/nui/text.md:759]: @safir-srbd , Hello, I hope you well too.
Yes, you are right. It should be ```SelectText```. 
I will update title.

Comment (safir-srbd) [reply docs/application/dotnet/guides/user-interface/nui/text.md:759]: @shrouqsabah Thank you for your response!

In that case, I have a heading suggestion for this section which would be more aligned with the rest of the titles in this document. 
What do you think of "Use clipboard and SelectText APIs" as the heading?
If it seems ok, please update it and we can merge this PR.

**Review by safir-srbd — APPROVED** [review]
@shrouqsabah Thank you for making the suggested change. All my comments have been addressed. 

I am approving this PR for merging.

**Pull request merged**
